File: c_plus/network_drive.cpp
#include <iostream>
#include <cstdlib>
#include <limits> // Connect header for numeric_limits

using namespace std;

void pauseWin() {
    /*
     * Only works in Windows
     */
    system("pause"); // Stops the console and waits for any key to be pressed
}

void pause() {
    cout << "Press Enter to continue...";
    cin.ignore(numeric_limits<streamsize>::max(), '\n'); // Clear the buffer
    cin.get(); // Waiting for Enter
}

void manageNetworkDrive() {
    string action;
    cout << "Enter 'SWITCH' to connect the network drive or 'OFF' to disconnect: ";
    cin >> action;

    // Connection data
    const string networkPath = R"(\\10.0.4.64\mikhailchm)";
    const string driveLetter = "Y:";

    if (action == "SWITCH") {
        // Connect a network drive
        const string command = "net use " + driveLetter + " " + networkPath;
        if (system(command.c_str()) == 0) {
            cout << "Network drive connected to " << driveLetter << endl;
            pause();
        } else {
            cerr << "Connection error!" << endl;
            pause();
        }
    } else {
        // Disconnect the network drive
        const string command = "net use " + driveLetter + " /delete /y";
        if (system(command.c_str()) == 0) {
            cout << "Network drive " << driveLetter << " disconnected." << endl;
            pause();
        } else {
            cerr << "Disconnection error!" << endl;
            pause();
        }
    }
}

int main() {
    manageNetworkDrive();
    return 0;
}

File: pyton/_tools/network_drive.py
import os


def manage_network_drive():
    # Request an action from the user
    action = input("Enter 'SWITCH' to connect the network drive or 'OFF' to disconnect: ").strip().upper()

    # Connection data
    network_path = r"\\10.0.4.64\mikhailchm"
    drive_letter = "Y:"

    if action == "SWITCH":
        # Connect a network drive
        try:
            os.system(f'net use {drive_letter} {network_path}')
            print(f"The network drive is connected to {drive_letter}")
        except Exception as e:
            print(f"Connection error: {e}")

    else:
        # Disconnect the network drive
        try:
            os.system(f'net use {drive_letter} /delete /y')
            print(f"Network disk {drive_letter} disabled")
        except Exception as e:
            print(f"Disconnect error: {e}")


if __name__ == "__main__":
    manage_network_drive()
